dashboard/src/components/UpgradeForm/UpgradeForm.tsx

```diff
@@ -24,172 +20,95 @@ interface IDeploymentFormProps {
   push: (location: string) => RouterAction;
   goBack: () => RouterAction;
   fetchChartVersions: (id: string) => Promise<IChartVersion[]>;
   getChartVersion: (id: string, chartVersion: string) => void;
-  clearRepo: () => void;
+  enableBasicForm: boolean;
 }
 
-interface IDeploymentFormState {
-  appValues?: string;
-  isDeploying: boolean;
+interface IUpgradeFormState {
+  appValues: string;
   valuesModified: boolean;
-  version: string;
+  isDeploying: boolean;
 }
 
-class UpgradeForm extends React.Component<IDeploymentFormProps, IDeploymentFormState> {
-  public state: IDeploymentFormState = {
-    appValues: this.props.appCurrentValues,
+class UpgradeForm extends React.Component<IUpgradeFormProps, IUpgradeFormState> {
+  public state: IUpgradeFormState = {
+    appValues: this.props.appCurrentValues || "",
     isDeploying: false,
     valuesModified: false,
-    version: this.props.appCurrentVersion,
   };
 
-  public componentDidMount() {
-    const { appCurrentVersion, chartName, fetchChartVersions, getChartVersion, repo } = this.props;
-    const chartID = `${repo}/${chartName}`;
-    fetchChartVersions(chartID);
-    getChartVersion(chartID, appCurrentVersion);
-  }
-
-  public componentDidUpdate(prevProps: IDeploymentFormProps) {
-    const { selected, appCurrentVersion, appCurrentValues } = this.props;
-    if (
-      selected.version &&
-      prevProps.selected.version &&
-      selected.version !== prevProps.selected.version
-    ) {
-      // Version has changed
-      if (selected.version.attributes.version === appCurrentVersion) {
-        // The user has selected back the original version, use the current values
-        if (!this.state.valuesModified) {
-          // Only update the default values if the user has not modify them
-          this.setState({ appValues: appCurrentValues });
-        }
-      }
-    }
-    if (selected.values && this.state.appValues && selected.values !== this.state.appValues) {
-      // Values has been modified either because the user has edit them
-      // or because the selected version is now different
-      if (!this.state.valuesModified) {
-        // Only update the default values if the user has not modify them
-        if (this.state.version !== appCurrentVersion) {
-          // Only use the default values if the version is not the original one
-          this.setState({ appValues: selected.values });
-        }
-      }
-    }
-  }
-
   public render() {
-    const { selected, namespace, releaseName, goBack } = this.props;
-    const { version, versions } = selected;
-    const { appValues } = this.state;
-    if (this.props.error) {
+    const { namespace, releaseName, error } = this.props;
+    if (error) {
       return (
-        <ErrorSelector
-          error={this.props.error}
-          namespace={namespace}
-          defaultRequiredRBACRoles={{ update: this.requiredRBACRoles() }}
-          action="update"
-          resource={`Application ${releaseName}`}
-        />
```

Comment:
So this is why the apprepository rbac role was still required for upgrades? Or it wasn't? It's not clear why this is removed with this change.

Reply:
Yes, after your changes, it's no longer required to have RBAC permissions to upgrade an app here. We forgot to remove this then and while I was refactoring I removed that.